src/DIRAC/Core/Utilities/StateMachine.py

```diff
@@ -30,19 +30,20 @@ class State(object):
   :param stateMap: it is a list (of strings) with the reachable states from this particular
                    status. If not defined, we assume there are no restrictions.
   :type stateMap: python:list
-  :param defState: default state used in case the next state it is not stateMap (not defined
+  :param defState: default state used in case the next state is not in stateMap (not defined
         or simply not there).
   :type defState: None or str
 
   """
 
-  def __init__(self, level, stateMap=list(), defState=None):
+  def __init__(self, level, stateMap=None, defState=None):
     """
     Constructor.
     """
 
     self.level = level
-    self.stateMap = stateMap
+    if stateMap is None:
```

Comment:
This if must go, otherwise `self.stateMap` isn't defined unless stateMap is None.